components/automate-cli/pkg/verifyserver/services/startmockserverservice/startmockserverservice.go

```diff
@@ -0,0 +1,265 @@
+package startmockserverservice
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/lib/logger"
+)
+
+// MockServerService provides functionality to start mock servers.
+type MockServerService struct {
+	MockServers []*models.Server
+	logger      logger.Logger
+}
+
+type IStartMockServersService interface {
+	StartMockServer(cfg *models.StartMockServerRequestBody) error
+	GetMockServers() []*models.Server
+}
+
+func New(logger logger.Logger) IStartMockServersService {
+	return &MockServerService{
+		logger: logger,
+	}
+}
+
+// func to get mock server list
+func (s *MockServerService) GetMockServers() []*models.Server {
+	return s.MockServers
+}
+
+// StartMockServer starts a mock server of the given type and port.
+func (servers *MockServerService) StartMockServer(cfg *models.StartMockServerRequestBody) error {
+	var myServer *models.Server
+	var err error
+	switch cfg.Protocol {
+	case constants.TCP:
+		myServer, err = servers.StartTCPServer(cfg.Port)
+	case constants.UDP:
+		myServer, err = servers.StartUDPServer(cfg.Port)
+	case constants.HTTPS:
```

Comment:
Check and validate the cert and key in case of HTTPS

Reply:
This is covered by error returned in `tls.X509KeyPair`